src/main/kotlin/org/rust/ide/formatter/impl/alignment.kt

```diff
@@ -0,0 +1,32 @@
+package org.rust.ide.formatter.impl
+
+import com.intellij.formatting.Alignment
+import com.intellij.openapi.util.Key
+import com.intellij.psi.tree.TokenSet
+import org.rust.ide.formatter.RustAlignmentStrategy
+import org.rust.ide.formatter.blocks.RustFmtBlock
+import org.rust.lang.core.psi.RustCompositeElementTypes.*
+
+/**
+ * Stores common alignment object for function declarations's parameters, return type & where clause.
+ */
+val PARAMETERS_ALIGNMENT: Key<Alignment> = Key.create("PARAMETERS_ALIGNMENT")
+
+fun RustFmtBlock.getAlignmentStrategy(): RustAlignmentStrategy = when (node.elementType) {
+    TUPLE_EXPR, TUPLE_TYPE, ARG_LIST, ENUM_TUPLE_ARGS ->
+        RustAlignmentStrategy.wrapCond { c, p -> !c.isBlockDelim(p) }
+
+    in PARAMS_LIKE ->
+        RustAlignmentStrategy.lazyWrapCond(
+            { this.getUserData(PARAMETERS_ALIGNMENT) },
+            { c, p -> !c.isBlockDelim(p) })
+
+    in FN_DECLS ->
+        RustAlignmentStrategy.lazyWrapFiltered(
+            { this.getUserData(PARAMETERS_ALIGNMENT) },
+            TokenSet.create(RET_TYPE, WHERE_CLAUSE))
```

Comment:
But such change has a cost of reducing versatility of `RustAlignmentStrategy`. Probably (but not 100% sure), `lazyWrap` will be also used to implement alignment for chain calls. Let's wait until all features of this PR will be implemented and then do proper cleanups